client/internal/dns/server.go

```diff
@@ -109,14 +118,16 @@ func newDefaultServer(ctx context.Context, wgInterface WGIface, dnsService servi
 		localResolver: &localResolver{
 			registeredMap: make(registrationMap),
 		},
-		wgInterface: wgInterface,
+		wgInterface:   wgInterface,
+		interfaceName: interfaceName,
+		wgAddr:        wgAddr,
 	}
 
 	return defaultServer
 }
 
 // Initialize instantiate host manager and the dns service
-func (s *DefaultServer) Initialize() (err error) {
+func (s *DefaultServer) Initialize(manager IosDnsManager) (err error) {
```

Comment:
In general to pass a nil to an exported function is bad practice. I do not know when can I set nil. And looks like that it depend from runtime (GOOS) variable when it will be used.  Hard to follow the logic when you try to write unit test.
So instead of a new function parameter just create a new constructor for the DefaultServer. We already have 2 because of the Android and feel free to add an iOS related.

Reply:
Yeah I didn't like that either. I will have a look at Android and see if I can do similar for iOS